src/html-imports.js

```diff
@@ -9,718 +9,754 @@
+          Path.replaceAttrUrl(v, base);
+      }
+    });
+  }
 
-function isImportLink(elt) {
-  return isLinkRel(elt, IMPORT_LINK_TYPE);
-}
+  function fixUrlsInTemplate(template, base) {
+    const content = template.content;
+    if (!content) { // Template not supported.
+      return;
+    }
+    const n$ = content.querySelectorAll('style, form[action], [src], [href], [url], [style]');
+    for (let i = 0; i < n$.length; i++) {
+      const n = n$[i];
+      if (n.localName == 'style') {
+        Path.resolveUrlsInStyle(n, base);
+      } else {
+        fixUrlAttributes(n, base);
+      }
+    }
+    fixUrlsInTemplates(content, base);
+  }
 
-function isLinkRel(elt, rel) {
-  return elt.localName === 'link' && elt.getAttribute('rel') === rel;
-}
+  function fixUrlsInTemplates(element, base) {
+    const t$ = element.querySelectorAll('template');
+    for (let i = 0; i < t$.length; i++) {
+      fixUrlsInTemplate(t$[i], base);
+    }
+  }
 
-function hasBaseURIAccessor(doc) {
-  return !! Object.getOwnPropertyDescriptor(doc, 'baseURI');
-}
+  const scriptType = 'import-script';
 
-/********************* vulcanize style inline processing  *********************/
-var attrs = ['action', 'src', 'href', 'url', 'style'];
-function fixUrlAttributes(element, base) {
-  for (var i=0, l=attrs.length, a, at, v; (i<l) && (a=attrs[i]); i++) {
-    at = element.attributes[a];
-    v = at && at.value;
-    if (v && (v.search(/({{|\[\[)/) < 0)) {
-      at.value = (a === 'style') ?
-        path.resolveUrlsInCssText(v, base) :
-        path.replaceAttrUrl(v, base);
+  function fixUrls(element, base) {
+    const n$ = element.querySelectorAll(importsSelectors);
+    for (let i = 0, l = n$.length, n; i < l && (n = n$[i]); i++) {
+      // Ensure we add load/error listeners before modifying urls or appending
+      // these to the main document.
+      whenElementLoaded(n);
+      if (n.href) {
+        n.setAttribute('href', Path.replaceAttrUrl(n.getAttribute('href'), base));
+      }
+      if (n.src) {
+        n.setAttribute('src', Path.replaceAttrUrl(n.getAttribute('src'), base));
+      }
+      if (n.localName == 'style') {
+        Path.resolveUrlsInStyle(n, base);
+      } else if (n.localName === 'script') {
+        if (n.textContent) {
+          n.textContent += `\n//# sourceURL=${base}`;
+        }
+        // NOTE: we override the type here, might need to keep track of original
+        // type and apply it to clone when running the script.
+        n.setAttribute('type', scriptType);
+      }
     }
+    fixUrlsInTemplates(element, base);
   }
-}
-
-function fixUrlsInTemplate(template, base) {
-  var content = template.content;
-  var n$ = content.querySelectorAll('style, form[action], [src], [href], [url], [style]');
-  for (var i=0; i < n$.length; i++) {
-    var n = n$[i];
-    if (n.localName == 'style') {
-      path.resolveUrlsInStyle(n, base);
-    } else {
-      fixUrlAttributes(n, base);
+
+  function fixDomModules(element, url) {
```

Comment:
Let's remove this function to eliminate the Polymer coupling, and instead add a general polyfill API that `dom-module` will call if the `HTMLImports` polyfill is loaded to find the original location of its import, something like:

```
HTMLImports.importForElement = function(el) {
  // walk up to find `<link rel="import">` and return that (or null)
}
```

Then, `dom-module` would need to gain a `resolveURL` API that calls `HTMLImports.importForElement` and returns a resolved URL.